src/everest/jobs/io/__init__.py

```diff
@@ -26,13 +26,13 @@ def load_data(filename):
     raise OSError(err_msg % (filename, str(yaml_err), str(json_err)))
 
 
-def _create_folders(filename):
+def _create_folders(filename: str) -> None:
     dirname = os.path.realpath(os.path.dirname(filename))
     if not os.path.isdir(dirname):
         os.makedirs(dirname)
 
 
-def safe_open(file, mode="r", **kwargs):
+def safe_open(file: Any, mode: str = "r", **kwargs: Any) -> Any:
```

Comment:
Shouldn't `file: Any` be `file: str`? 
Also as the return type you could try `TextIO` `from typing import TextIO`